### PR TITLE
🌱 Minor improvements to godoc, code style in builder pkg

### PR DESCRIPTION
**Pull request by sbueringer**

### CODE REVIEW

File: pkg/builder/controller.go [hunk 1]
```diff
@@ -344,12 +340,13 @@ func (blder *Builder) doController(r reconcile.Reconciler) error {
 	globalOpts := blder.mgr.GetControllerOptions()
 
 	ctrlOptions := blder.ctrlOptions
-	if ctrlOptions.Reconciler == nil {
-		ctrlOptions.Reconciler = r
+	if ctrlOptions.Reconciler != nil {
+		return errors.New("reconciler was already set via WithOptions()")
```
Comment (sbueringer) [inline pkg/builder/controller.go:344]: I think it makes sense to return an error when two reconcilers are set via the buidler and not just silently overwrite

File: pkg/builder/controller.go [hunk 2]
```diff
@@ -344,12 +340,15 @@ func (blder *Builder) doController(r reconcile.Reconciler) error {
 	globalOpts := blder.mgr.GetControllerOptions()
 
 	ctrlOptions := blder.ctrlOptions
+	if ctrlOptions.Reconciler != nil && r != nil {
```
Comment (sbueringer) [inline pkg/builder/controller.go:343]: I think it makes sense to return an error when two reconcilers are set via the buidler and not just silently pick one

(I can move this to a separate PR if preferred)

File: pkg/builder/controller.go [hunk 3]
```diff
@@ -311,19 +310,16 @@ func (blder *Builder) doWatch() error {
 		return errors.New("there are no watches configured, controller will never get triggered. Use For(), Owns() or Watches() to set them up")
 	}
 	for _, w := range blder.watchesInput {
-		allPredicates := append([]predicate.Predicate(nil), blder.globalPredicates...)
```
Comment (inteon) [inline pkg/builder/controller.go:314]: I liked this better, it explicitly makes sure we are not editing the `globalPredicates` value.

Comment (sbueringer) [reply pkg/builder/controller.go:314]: My bad. You're right

Comment (sbueringer) [reply pkg/builder/controller.go:314]: If I got it right we should also do this in l.277

Comment (sbueringer) [reply pkg/builder/controller.go:314]: Thx for catching this!

File: pkg/builder/controller.go [hunk 4]
```diff
@@ -274,7 +274,8 @@ func (blder *Builder) doWatch() error {
 		}
 		src := source.Kind(blder.mgr.GetCache(), obj)
 		hdler := &handler.EnqueueRequestForObject{}
-		allPredicates := append(blder.globalPredicates, blder.forInput.predicates...)
+		allPredicates := append([]predicate.Predicate(nil), blder.globalPredicates...)
```
Comment (sbueringer) [inline pkg/builder/controller.go:277]: Just aligning to the way we do it below. I played a bit around with it, I don't think it makes a difference

**Review by alvaroaleman — APPROVED** [review]

**Comment by sbueringer** [comment]
Should be an independent flake

/retest

**Pull request merged**